simba/report.py

```diff
@@ -267,6 +291,268 @@ def generate(schedule, scenario, args):
+        fig, ax = plt.subplots()
+
+        agg_ts = aggregate_timeseries(scenario, gc)
+        headers = [
+            "grid supply [kW]",
+            "fixed load [kW]",
+            "local generation [kW]",
+            "sum CS power [kW]",
+            "battery power [kW]",
+            "bat. stored energy [kWh]",
+        ]
+
+        time_index = agg_ts["header"].index("time")
+        time_values = [row[time_index] for row in agg_ts["timeseries"]]
+
+        for index, header in enumerate(headers):
+            try:
+                header_index = agg_ts["header"].index(header)
+                header_values = [row[header_index] for row in agg_ts["timeseries"]]
+
+                if header == "bat. stored energy [kWh]":
+                    ax2 = ax.twinx()
+                    ax2.set_ylabel("Power in kWh")
+                    next_color = plt.rcParams['axes.prop_cycle'].by_key()["color"][index]
+                    ax2.plot(time_values, header_values, label=header, c=next_color,
+                             linestyle="dashdot")
+                    ax2.legend()
+                    ax2.tick_params(axis='x', rotation=30)
+                    fig.set_size_inches(8, 4.8)
+                else:
+                    ax.plot(time_values, header_values, label=header)
+            except ValueError:
+                continue
+        try:
+            ax2.set_ylim(ax.get_ylim())
+        except NameError:
+            continue
+
+        ax.legend()
+        ax.tick_params(axis='x', rotation=30)
+        ax.set_ylabel("Power in kW")
+        ax.set_title(f"Power: {gc}")
+        ax.grid(color='gray', linestyle='-')
+
+        gc_cleaned = gc.replace("/", "").replace(".", "")
+        plt.savefig(extended_plots_path / f"{gc_cleaned}_power_time_overview.png")
+        plt.close(fig)
+
+
+def active_rotations(extended_plots_path, scenario, schedule):
+    """Generate a plot where the number of active rotations is shown.
+
+    :param extended_plots_path: Path argument which combines args.results_directory
+                                and string 'extended_plots'
+    :type extended_plots_path: PosixPath
+    :param schedule: Driving schedule for the simulation. schedule.rotations are used
+    :type schedule: eBus-Toolbox.Schedule
+    :param scenario: Provides the data for the grid connectors over time.
+    :type scenario: spice_ev.Scenario
+    """
+    ts = [
+        scenario.start_time if i == 0 else scenario.start_time + scenario.interval * i
+        for i in range(scenario.n_intervals)
+    ]
+    active_vehicles = [len(scenario.components.vehicles)] * scenario.n_intervals
+    depot_stations = [
+        station for station in scenario.components.grid_connectors
+        if schedule.stations[station]["type"] == "deps"
+    ]
+    for station_name in depot_stations:
+        station_ts = scenario.connChargeByTS[station_name]
+        for index, step in enumerate(station_ts):
+            active_vehicles[index] -= len(step)
```

Comment:
See comment 258ff. Consider moving this segment to its own util or schedule function (suggestion: count_active_vehicles)